### DISCUSSION TITLE
Background scheduler didn't run for a long period (150s), results might be inaccurate

### DISCUSSION THREAD

**Discussion by milad7093** [connectomicslab/connectomemapper3#221] [Q&A]
Dear Sebastian 

I ran the cmp3 for 8 subjects using parallel run on my PC ( ubuntu 20.04, 24 cores and 16 GB Ram). After running the program for about 4 hours , There were some  warning massages in the terminal as follow: 

connectomemapper3_docker  "/media/mn/adata_hm800/imaging_data/backup_group_HC_prodromal/dwi" "/media/mn/adata_hm800/imaging_data/backup_group_HC_prodromal/dwi/derivatives"  participant --participant_label 56886 57484 60006 60023  --session_label 1   --fs_license "/media/mn/adata_hm800/imaging_data/backup_group_HC_prodromal/dwi/code/license.txt"      --config_dir "/media/mn/adata_hm800/imaging_data/backup_group_HC_prodromal/dwi/code"  --track_carbon_footprint  --anat_pipeline_config "ref_anatomical_config.json"  --dwi_pipeline_config "ref_diffusion_config.json" --number_of_participants_processed_in_parallel 4
... cmd: docker run -t --rm -u $(id -u):$(id -g) -v /media/mn/adata_hm800/imaging_data/backup_group_HC_prodromal/dwi:/bids_dir -v /media/mn/adata_hm800/imaging_data/backup_group_HC_prodromal/dwi/derivatives:/output_dir -v /media/mn/adata_hm800/imaging_data/backup_group_HC_prodromal/dwi/code:/config -v /media/mn/adata_hm800/imaging_data/backup_group_HC_prodromal/dwi/code/license.txt:/bids_dir/code/license.txt sebastientourbier/connectomemapper-bidsapp:v3.0.4 /bids_dir /output_dir participant --participant_label 56886 57484 60006 60023 --session_label 1 --anat_pipeline_config /config/ref_anatomical_config.json --dwi_pipeline_config /config/ref_diffusion_config.json --fs_license /bids_dir/code/license.txt --number_of_participants_processed_in_parallel 4 
Execution of job "BaseEmissionsTracker._measure_power (trigger: interval[0:00:15], next run at: 2022-08-23 12:30:07 +0430)" skipped: maximum number of running instances reached (1)
Run time of job "BaseEmissionsTracker._measure_power (trigger: interval[0:00:15], next run at: 2022-08-23 12:30:07 +0430)" was missed by 0:00:22.641816
Run time of job "BaseEmissionsTracker._measure_power (trigger: interval[0:00:15], next run at: 2022-08-23 12:31:52 +0430)" was missed by 0:00:07.017509
Run time of job "BaseEmissionsTracker._measure_power (trigger: interval[0:00:15], next run at: 2022-08-23 12:32:07 +0430)" was missed by 0:00:01.201210
CODECARBON : Background scheduler didn't run for a long period (150s), results might be inaccurate

As you can see, This massage warns me regarding  results. What is your suggestion? Can i ignore it and trust all outputs? 

Thanks in advance to your helps,

Best regards,

-Milad

**Comment by sebastientourbier** [comment]
Dear Milad,

In fact, this comes from the carbon footprint estimator that seems to not behave well in the case of parallel execution... So I would not trust the carbon footprint estimates. But it is not related to the CMP3 pipeline itself and so, you should not worry about CMP3 outputs.

Best,
Seb
